### ISSUE TITLE
[SL-3389] [Feature] MetricFlow doesn't support `DBT_PROJECT_DIR` environment variable

### ISSUE THREAD

**Issue by mercedes-wu-wandb** [dbt-labs/metricflow#1591]
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

commands from dbt project level:
mf health-checks

output:
```
AttributeError: 'Namespace' object has no attribute 'PROFILES_DIR'
```

### Expected Behavior

```
✔ Health checks completed.
• ✅ SqlEngine.BIGQUERY - SELECT 1: Success!
```

### Steps To Reproduce

1. Set up a dbt project with a valid profiles.yml.
2. Set the DBT_PROJECT_DIR environment variable to point to the project directory:
`export DBT_PROJECT_DIR=/path/to/dbt/project`
3. Set the DBT_PROFILES_DIR environment variable to point to the profiles 
`export DBT_PROFILES_DIR=/path/to/profiles/directory`
4. Run the following command from the dbt directory:
`mf health-checks`

### Environment

```markdown
- OS:Darwin Kernel Version 23.6.0
- Python: 3.12.3
- dbt: 1.8.9
```

### Which database adapter are you using with dbt?

bigquery

### Additional Context

I am trying to run metricflow on dbt-core

<sub>[SL-3389](https://linear.app/dbt-labs/issue/SL-3389/[bug]-metricflow-doesnt-support-dbt-project-dir-environment-variable)</sub>

**Comment by dbeatty10** [comment]
Transferred from https://github.com/dbt-labs/dbt-core/issues/11179 to https://github.com/dbt-labs/metricflow/issues/1591.

**Comment by plypaul** [comment]
@mercedes-wu-wandb To clarify, `mf` works for you when run it from the dbt project directory?

**Comment by mercedes-wu-wandb** [comment]
Yes `mf` works in the dbt project directory but even with my environment variable set to `DBT_PROJECT_DIR` it does not seem to recognize it

```
analytics-py3.12➜  analytics git:(semantic-layer) mf health-checks
‼️ Warning: A new version of the MetricFlow CLI is available.
💡 Please update to version 0.8.1, released 2025-02-23 09:44:59 by running:
        $ pip install --upgrade dbt-metricflow

❌ Unable to locate 'dbt_project.yml' in the current directory
In order to run the MetricFlow CLI, you must be running in the root directory of a working dbt project.
Please check out `https://docs.getdbt.com/reference/commands/init` if you want to get started on building a dbt project.
analytics-py3.12➜  analytics git:(semantic-layer) cd dbt
analytics-py3.12➜  dbt git:(semantic-layer) mf health-checks
‼️ Warning: A new version of the MetricFlow CLI is available.
💡 Please update to version 0.8.1, released 2025-02-23 09:44:59 by running:
        $ pip install --upgrade dbt-metricflow

✔ Health checks completed.
• ✅ SqlEngine.BIGQUERY - SELECT 1: Success!
```

**Comment by plypaul** [comment]
It should be straightforward to add support for `DBT_PROJECT_DIR`, but did you see documentation that allows for that environment variable for the `mf` CLI? Looking to  double check the classification of this as a bug vs. feature.

**Comment by mercedes-wu-wandb** [comment]
Hey @plypaul, so far i have not so I'm happy to change this to issue to a feature if that's helpful!

**Comment by plypaul** [comment]
Yes, that would be great. Thanks!

**Comment by plypaul** [comment]
There was some relevant prior work that has already been done, and I'm aiming to get a PR out this week.

**Comment by mercedes-wu-wandb** [comment]
awesome, thanks a lot for the updates and looking forward to the PR!

**Comment by mercedes-wu-wandb** [comment]
Hi @plypaul ! Just wondering if there were any updates on this feature, thanks a ton for all the help so far!

**Comment by plypaul** [comment]
To give an update, getting this out for release required a large number of changes to the project's CLI test framework as we want to have good test coverage for this feature. In addition, as I was getting ramped up on the CLI implementation, I saw there were a few CLI bugs (e.g. display of numeric types) that would be beneficial to have in the release. Right now, the feature is in the review pipeline but at the bottom of a stack of changes:

[https://github.com/dbt-labs/metricflow/pull/1689]()
[https://github.com/dbt-labs/metricflow/pull/1690]()
[https://github.com/dbt-labs/metricflow/pull/1691]()
[https://github.com/dbt-labs/metricflow/pull/1692]()
[https://github.com/dbt-labs/metricflow/pull/1693]()
[https://github.com/dbt-labs/metricflow/pull/1694]()
[https://github.com/dbt-labs/metricflow/pull/1695]()
[https://github.com/dbt-labs/metricflow/pull/1696]()
[https://github.com/dbt-labs/metricflow/pull/1697]()
[https://github.com/dbt-labs/metricflow/pull/1698]()
[https://github.com/dbt-labs/metricflow/pull/1699]()
[https://github.com/dbt-labs/metricflow/pull/1700]()
[https://github.com/dbt-labs/metricflow/pull/1701]()
[https://github.com/dbt-labs/metricflow/pull/1702]() `<-- PR for this issue`

The PR for this feature is relatively small, but partly due to the work that was included by prior PRs in the stack.

We're looking to get these reviewed and merged, and then I can cut a release. While there a number of changes, I think we can get it out this week.

**Comment by mercedes-wu-wandb** [comment]
thanks a ton for the detailed update! really appreciate the help and looking forward to the release!

**Comment by plypaul** [comment]
@mercedes-wu-wandb FYI, this went out in `dbt-metricflow==0.8.2`. Please feel free to reopen this issue if there are any problems.

**Comment by mercedes-wu-wandb** [comment]
Hey I'm so sorry @plypaul i forgot to respond back - this worked great and thank you so much for the help!

**Comment by plypaul** [comment]
Glad to hear!
